config/routes.rb

```diff
@@ -21,6 +21,8 @@
 
   api :v1, default: true do
     resources :exercises do
+      post :save_image, on: :member
```

Comment:
The attachment isn't necessarily an "image" per se.  Currently it could be a PDF.  Later one could imagine letting users upload code snippets, or other short documents.  

So I would prefer for the route to not have "image" in it.  In fact, you could do

```ruby
...
  resources :exercises do
    resources :attachments, shallow: true, only: :create
```

To get `POST /exercises/:id/attachments`